### ISSUE TITLE
Normalize BOM for sha1sum

### ISSUE THREAD

**Issue by phaza** [dotnet/sourcelink#237]
Our nuget package contains a file with sha1sum `56fc41fa153c3d0d9f2d2c90f7dddaeb98d5a74f`.
Visual studio downloads the correct file from gitlab, but thinks it's not a match, since it seems to do the sha1sum without BOM (which is ironic given how VS insists on BOM for new files). The sha1sum of the file downloaded by Visual studio without BOM is `236030d57173648386efe52a39888560a6e568db`.

Fairly annoying.

`<PackageReference Include="Microsoft.SourceLink.GitLab" Version="1.0.0-beta-63127-02" PrivateAssets="All" />`
The nuget package is made in a linux docker image (microsoft/dotnet:2.1-sdk), on a linux/gentoo host.

Visual studio:
![image](https://user-images.githubusercontent.com/4553/51846403-d0751780-2319-11e9-9aa0-66d12c510fd7.png)

Command line testing sha1sum:
![image](https://user-images.githubusercontent.com/4553/51846425-dc60d980-2319-11e9-945c-4a2d680ff2dd.png)

**Comment by tmat** [comment]
That's odd. The debugger is supposed to calculate the hash on the binary stream, not on the textual representation. There should be no special treatment of BOM or any other text encoding artifacts.

**Comment by tmat** [comment]
@chuckries

**Comment by chuckries** [comment]
@phaza  is your GitLab repo public or private?

**Comment by chuckries** [comment]
@phaza Can you please answer the following:
1. Is your GitLab repo public or private?
2. Which version of Visual Studio are you using?
3. It appears that there is a cached version of the file in your "%LOCALAPPDATA%\SourceServer" directory (you blacked it out in the screenshot you attached). Please try deleting the cached version and try again.

What I think is happening is that somebody is dropping the BOM while downloading the file from GitLab. Either GitLab is serving it up without the BOM or we are failing to preserve the BOM when we write the contents of the Source Link repsonse to disk.

**Comment by phaza** [comment]
Terrible sorry, I found the error. We run a gitlab private instance, and I wrote a reverse proxy in front to handle Http Basic->OAuth authentication, since Basic auth is not supported by gitlab. The mistake was that in reading data from gitlab with HttpClient, I was using `ReadAsStringAsync`, not realizing that it actually [removes the BOM](https://github.com/dotnet/corefx/blob/28e216680a990ce4a58798aa0d2026c9d3e1f148/src/System.Net.Http/src/System/Net/Http/HttpContent.cs#L165-L245).

Switching to reading the response as a byte array and manually converting to string solved the problem.

Sorry for wasting your time, and thank you for trying to help.
